### PR TITLE
Minor refactorings and Rename NeuralNet to MLP

### PR DESCRIPTION
**Pull request by hyeok9855**
This commit contains:
- several incremental refactorings
- remaning `NeuralNet` to `MLP`, considering the broader concept of neural network.

### CODE REVIEW

**Comment by josephdviviano** [comment]
I think you just need to resolve conflict with recent updates to `master`

File: src/gfn/containers/replay_buffer.py
Comment (josephdviviano) [inline src/gfn/containers/replay_buffer.py:43]: good catch

File: src/gfn/env.py
Comment (josephdviviano) [inline src/gfn/env.py:278]: thank you

File: src/gfn/gym/helpers/preprocessors.py
Comment (josephdviviano) [inline src/gfn/gym/helpers/preprocessors.py:38]: This one i'm not sure of - why did you feel compelled to remove it?

File: src/gfn/utils/__init__.py
Comment (josephdviviano) [inline src/gfn/utils/__init__.py:1]: I don't mind this change but I'm wondering what your motivation was here?

File: tutorials/examples/train_hypergrid.py
Comment (josephdviviano) [inline tutorials/examples/train_hypergrid.py:203]: Nice catch :)

File: tutorials/examples/train_hypergrid_simple.py
Comment (josephdviviano) [inline tutorials/examples/train_hypergrid_simple.py:61]: Why did you remove this (and `batch_size` and `n_iterations`)?

**Review by josephdviviano — CHANGES_REQUESTED** [review]
Great stuff! A few style / other questions (nothing major but good to get on the same page before approval).

File: src/gfn/gym/helpers/preprocessors.py
Comment (hyeok9855) [reply src/gfn/gym/helpers/preprocessors.py:38]: Since,
1. It was not used!
2. The parent class `Preprocessor` does not have it as a parameter

File: src/gfn/utils/__init__.py
Comment (hyeok9855) [reply src/gfn/utils/__init__.py:1]: Previously, `NeuralNet` and `Tabular` were imported like this:

```python
from utils import NeuralNet, Tabular
```

However, I found this a bit odd since `utils` contains several submodules, each with its own specific functionality. So, I thought it made more sense to import them with the submodule specified, like this:

```python
from utils.modules import NeuralNet, Tabular
```

File: tutorials/examples/train_hypergrid_simple.py
Comment (hyeok9855) [reply tutorials/examples/train_hypergrid_simple.py:61]: Sorry, I didn't mean to include this change initially—it was unintentionally committed with the whole file. However, after taking another look, I think there’s room for improvement, like using argparse instead of relying on global variables. I’ll push this update in the next commits!

Comment (josephdviviano) [reply tutorials/examples/train_hypergrid_simple.py:57]: Totally agree with that!!

File: src/gfn/utils/__init__.py
Comment (josephdviviano) [reply src/gfn/utils/__init__.py:1]: Yes I see, makes sense!

What I'm beginning to think is we will need a whole `modules` directory. Containing both the core GFN modules, plus these auxiliary modules, because I feel like we are going to get more and more of them over time - and the distinction between `modules.py` and `utils/modules.py` isn't obvious to new users :)

File: src/gfn/gym/helpers/preprocessors.py
Comment (josephdviviano) [reply src/gfn/gym/helpers/preprocessors.py:38]: Oh no!  Great catch if so. Can you ensure this is truly not needed across the entire library? I don't mind if we keep this as part of the current PR.

Screenshot 2024-10-28 125646

Comment (hyeok9855) [reply src/gfn/gym/helpers/preprocessors.py:38]: I've checked them thoroughly. The change has no problem, and the other parts seem to have no problem.

Comment (josephdviviano) [reply src/gfn/gym/helpers/preprocessors.py]: Thanks!

**Review by josephdviviano — APPROVED** [review]
lgtm!! thanks

**Pull request merged**
